examples/adapters/modeling.py

```diff
@@ -0,0 +1,166 @@
+    
+    self.up_project = tf.keras.layers.Dense(
+                        input_size,
+                        kernel_initializer=tf.keras.initializers.TruncatedNormal(stddev=0.02),
+                        bias_initializer="zeros",
+                        name="feedforward_upproject")
+    
+    AdapterModule.__instance = self
+
+  def call(self, inputs, **kwargs):
+
+    output = self.down_project(inputs)
+    output = self.non_linearity(output)
+    output = self.up_project(output)
+    output = output + inputs
+    return output
+
+
+
+class TFBertLayer(tf.keras.layers.Layer):
+    def __init__(self, pretrained_self_attn, pretrained_self_dense, pretrained_self_ln, pretrained_intermediate, pretrained_out_dense, pretrained_out_ln, config, **kwargs):
+        super().__init__(**kwargs)
+        self.attention = TFBertAttention(pretrained_self_attn, pretrained_self_dense, pretrained_self_ln, config, name="attention")
+        # shared new task layer normalization in each layer
+        # self.TaskLayerNorm = tf.keras.layers.LayerNormalization(epsilon=config.layer_norm_eps, name="TaskLayerNorm")
+        self.intermediate = copy.deepcopy(pretrained_intermediate)
+        self.bert_output = TFBertOutput(pretrained_out_dense, pretrained_out_ln, config, name="output")
+
+
+    def call(self, inputs, training=False):
+        hidden_states, attention_mask, head_mask, output_attentions = inputs
+
+        attention_outputs = self.attention(
+            [hidden_states, attention_mask, head_mask, output_attentions], training=training
+        )
+        attention_output = attention_outputs[0]
+        # attention_output = self.TaskLayerNorm(attention_output)
+        intermediate_output = self.intermediate(attention_output)
+        layer_output = self.bert_output([intermediate_output, attention_output], training=training)
+        # layer_output = self.TaskLayerNorm(layer_output)
+        outputs = (layer_output,) + attention_outputs[1:]  # add attentions if we output them
+        return outputs
+
+
+
+class TFBertAttention(tf.keras.layers.Layer):
+    def __init__(self, pretrained_self_attn, pretrained_self_dense, pretrained_self_ln, config, **kwargs):
+        super().__init__(**kwargs)
+        self.self_attention = copy.deepcopy(pretrained_self_attn)
+        self.dense_output = TFBertSelfOutput(pretrained_self_dense, pretrained_self_ln, config, name="output")
+
+    def prune_heads(self, heads):
+        raise NotImplementedError
+
+    def call(self, inputs, training=False):
+        input_tensor, attention_mask, head_mask, output_attentions = inputs
+
+        self_outputs = self.self_attention(
+            [input_tensor, attention_mask, head_mask, output_attentions], training=training
+        )
+        attention_output = self.dense_output([self_outputs[0], input_tensor], training=training)
+        outputs = (attention_output,) + self_outputs[1:]  # add attentions if we output them
+        return outputs
```

Comment:
You can import these from `tf_modeling_bert.py`, no?

Reply:
Actually, the part of the code that has commented out in the TFBertLayer class indicates new layer norms that according to the paper, are added to each layer of the transformer after each adapter module. However, since these layers are not used in practice (even in the [original source code](https://github.com/google-research/adapter-bert/blob/master/modeling.py)), this part of the code could be removed.